### PR TITLE
add cloud9 instruction in the drift-detection

### PR DESCRIPTION
**Pull request by shubham-amaz**
## What does this PR do and why?

add cloud9 instruction in the drift-detection

## PR acceptance checklist

This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] Added/Updated documentation if applicable.
- [x] Pre-commit checks passed.
- [x] Lint and Nag checks passed.
- [ ] If releasing a new version, have you bumped the version `make version part=<major|minor|patch>`?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/intermediate/operations/drift-detection/index.md [hunk 1]
```diff
@@ -60,13 +60,33 @@ Resources:
...
+
+   :::::tabs{variant="container"}
+	::::tab{id="cloud9" label="Cloud9"}
+	1. In the **Cloud9 terminal** navigate to `code/workspace/drift-detection`:
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    cd cfn101-workshop/ccode/workspace/drift-detection
+    :::
+    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    aws cloudformation create-stack --stack-name drift-detection-workshop --template-body file://drift-detection-workshop.yaml
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:72]:
```suggestion
    aws cloudformation create-stack --stack-name cfn-workshop-drift-detection --template-body file://drift-detection-workshop.yaml
```

File: content/intermediate/operations/drift-detection/index.md [hunk 2]
```diff
@@ -60,13 +60,33 @@ Resources:
...
+    :::
+    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/drift-detection-workshop/739fafa0-e4d7-11ed-a000-12d9009553ff"
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:76]:
```suggestion
    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-drift-detection/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

File: content/intermediate/operations/drift-detection/index.md [hunk 3]
```diff
@@ -60,13 +60,33 @@ Resources:
...
+    :::
+    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
+    ::::
+    ::::tab{id="local" label="Local Development"}
+   1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
+   1. From **Create stack**, choose **With new resources (standard)**.
+   1. From **Specify template**, choose **Upload a template file**. Choose the `drift-detection-workshop.yaml` template mentioned earlier, and then choose **Next**.
+   1. Enter a stack name. For example, specify `drift-detection-workshop`. Choose **Next**.
+   1. In **Configure stack options**, choose **Next**.
+   1. Choose **Create stack**.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:86]:
```suggestion
   1. Choose **Submit**.
```

File: content/intermediate/operations/drift-detection/index.md [hunk 4]
```diff
@@ -60,13 +60,33 @@ Resources:
...
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/drift-detection-workshop/739fafa0-e4d7-11ed-a000-12d9009553ff"
+    :::
+    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
+    ::::
+    ::::tab{id="local" label="Local Development"}
+   1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
+   1. From **Create stack**, choose **With new resources (standard)**.
+   1. From **Specify template**, choose **Upload a template file**. Choose the `drift-detection-workshop.yaml` template mentioned earlier, and then choose **Next**.
+   1. Enter a stack name. For example, specify `drift-detection-workshop`. Choose **Next**.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:84]:
```suggestion
   1. Enter a stack name. For example, specify `cfn-workshop-drift-detection`. Choose **Next**.
```

File: content/intermediate/operations/drift-detection/index.md [hunk 5]
```diff
@@ -60,13 +60,33 @@ Resources:
...
+	1. In the **Cloud9 terminal** navigate to `code/workspace/drift-detection`:
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    cd cfn101-workshop/ccode/workspace/drift-detection
+    :::
+    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    aws cloudformation create-stack --stack-name drift-detection-workshop --template-body file://drift-detection-workshop.yaml
+    :::
+    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:75]:
```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

File: content/intermediate/operations/drift-detection/index.md [hunk 6]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
 1. In your favourite text editor, open the template file for the workshop.
 2. Modify `Queue1`'s `MessageRetentionPeriod` to match the value shown in the **Actual** column of the drift details page you saw in the previous step. In your template, set the value for `MessageRetentionPeriod` to `172800`, that is the number of seconds in `2` days.
 3. Save the template file.
+1. It’s now time to update your stack! Follow steps below:
+
 4. Navigate to the [CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:154]:
```suggestion
```

File: content/intermediate/operations/drift-detection/index.md [hunk 7]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
 5. Choose your stack as before.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:155]:
```suggestion
```

File: content/intermediate/operations/drift-detection/index.md [hunk 8]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
+
+   :::::tabs{variant="container"}
+	::::tab{id="cloud9" label="Cloud9"}
+	1. In the **Cloud9 terminal** navigate to `code/workspace`:
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    cd cfn101-workshop/code/workspace
+    :::
+    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    aws cloudformation update-stack --stack-name cfn-workshop-s3 --template-body file://template-and-stack.yaml
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:167]:
```suggestion
    aws cloudformation update-stack --stack-name cfn-workshop-drift-detection --template-body file://template-and-stack.yaml
```

File: content/intermediate/operations/drift-detection/index.md [hunk 9]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
+    :::
+    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:170]:
```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

File: content/intermediate/operations/drift-detection/index.md [hunk 10]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
+    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-s3/739fafa0-e4d7-11ed-a000-12d9009553ff"
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:171]:
```suggestion
    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-drift-detection/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

File: content/intermediate/operations/drift-detection/index.md (hunk 9 above)
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:170]:
```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

File: content/intermediate/operations/drift-detection/index.md [hunk 11]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
+   1. Log in to the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new browser tab.
+   1. Navigate to the [CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
+   1. Choose your stack as before.
+   1. Choose **Update**.
+   1. Choose **Replace current template**, then choose **Upload a template file**.
+   1. Use **Choose file** to select your updated template file.
+   1. Choose **Next**.
+   1. On the stack details page, choose **Next**.
+   1. On the stack options page, choose **Next**.
+   1. Choose **Update Stack**.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:185]:
```suggestion
   1. Choose **Submit**.
```

File: content/intermediate/operations/drift-detection/index.md [hunk 12]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
-14. Choose the **Stack info** tab.
-15. From **Stack actions**, choose **Detect Drift**.
-16. Wait a few seconds for drift detection to complete.
-17. You should now see that the drift status is `IN_SYNC`, showing that the template and resource match.
+
+1. It’s now time to update your stack! Follow steps below:
+
+   :::::tabs{variant="container"}
+	::::tab{id="cloud9" label="Cloud9"}
+	1. In the **Cloud9 terminal** navigate to `code/workspace`:
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:161]:
```suggestion
	1. In the **Cloud9 terminal** ensure your working directory is `code/workspace/drift-detection`:
```

File: content/intermediate/operations/drift-detection/index.md [hunk 13]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    cd cfn101-workshop/code/workspace
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:163]:
```suggestion
    cd cfn101-workshop/code/workspace/drift-detection
```

File: content/intermediate/operations/drift-detection/index.md [hunk 14]
```diff
@@ -129,20 +149,47 @@ You will now update the template to match the new state of the resource and brin
...
+    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    aws cloudformation update-stack --stack-name cfn-workshop-s3 --template-body file://template-and-stack.yaml
+    :::
+    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-s3/739fafa0-e4d7-11ed-a000-12d9009553ff"
+    :::
+    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
+    ::::
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:174]: Need to add in the stack drift check
```suggestion
 1. Choose the **Stack info** tab.
   1. From **Stack actions**, choose **Detect Drift**.
   1. Wait a few seconds for drift detection to complete.
   1. You should now see that the drift status is `IN_SYNC`, showing that the template and resource match.
    ::::
```

File: content/intermediate/operations/drift-detection/index.md [hunk 15]
```diff
@@ -177,13 +224,36 @@ Resources:
...
-4. Enter a stack name, for example `drift-detection-challenge` and choose **Next**.
-5. In **Configure Stack Options**, choose **Next**.
-6. In the next page, choose **Create stack**.
-7. Once the stack is created, select the `drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
+
+   :::::tabs{variant="container"}
+	::::tab{id="cloud9" label="Cloud9"}
+	1. In the **Cloud9 terminal** navigate to `code/workspace`:
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    cd cfn101-workshop/code/workspace
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:232]:
```suggestion
    cd cfn101-workshop/code/workspace/drift-detection
```

File: content/intermediate/operations/drift-detection/index.md [hunk 16]
```diff
@@ -177,13 +224,36 @@ Resources:
...
-2. From **Create stack**, choose **With new resources (standard)**.
-3. From **Specify template**, choose **Upload a template file**, upload the `drift-detection-challenge.yaml` file and choose **Next**.
-4. Enter a stack name, for example `drift-detection-challenge` and choose **Next**.
-5. In **Configure Stack Options**, choose **Next**.
-6. In the next page, choose **Create stack**.
-7. Once the stack is created, select the `drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
+
+   :::::tabs{variant="container"}
+	::::tab{id="cloud9" label="Cloud9"}
+	1. In the **Cloud9 terminal** navigate to `code/workspace`:
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:230]:
```suggestion
	1. In the **Cloud9 terminal** ensure your working directory is `code/workspace/drift-detection`:
```

File: content/intermediate/operations/drift-detection/index.md [hunk 17]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    cd cfn101-workshop/code/workspace
+    :::
+    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    aws cloudformation update-stack --stack-name cfn-workshop-s3 --template-body file://template-and-stack.yaml
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:236]:
```suggestion
    aws cloudformation update-stack --stack-name cfn-workshop-drift-detection-challenge --template-body file://drift-detection-challenge.yaml
```

File: content/intermediate/operations/drift-detection/index.md [hunk 18]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+    :::
+    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:239]:
```suggestion
    :::code{language=shell showLineNumbers=false showCopyAction=false}
```

File: content/intermediate/operations/drift-detection/index.md [hunk 19]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-s3/739fafa0-e4d7-11ed-a000-12d9009553ff"
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:240]:
```suggestion
    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-drift-detection-challenge/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

File: content/intermediate/operations/drift-detection/index.md [hunk 20]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+    :::
+    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
+    ::::
+    ::::tab{id="local" label="Local Development"}
+   1. Log in to the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new browser tab.
+   1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
+   1. From **Create stack**, choose **With new resources (standard)**.
+   1. From **Specify template**, choose **Upload a template file**, upload the `drift-detection-challenge.yaml` file and choose **Next**.
+   1. Enter a stack name, for example `drift-detection-challenge` and choose **Next**.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:249]:
```suggestion
   1. Enter a stack name, for example `cfn-workshop-drift-detection-challenge` and choose **Next**.
```

File: content/intermediate/operations/drift-detection/index.md [hunk 21]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+   1. In **Configure Stack Options**, choose **Next**.
+   1. In the next page, choose **Create stack**.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:251]:
```suggestion
   1. In the next page, choose **Submit**.
```

File: content/intermediate/operations/drift-detection/index.md [hunk 22]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+   1. Once the stack is created, select the `drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:252]:
```suggestion
   1. Once the stack is created, select the `cfn-workshop-drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
```

File: content/intermediate/operations/drift-detection/index.md [hunk 23]
```diff
@@ -177,13 +224,36 @@ Resources:
...
+    1. Use the AWS CLI to create the stack. The required parameters `--stack-name` and `--template-body` have been pre-filled for you.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    aws cloudformation update-stack --stack-name cfn-workshop-s3 --template-body file://template-and-stack.yaml
+    :::
+    1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
+    :::code{language=shell showLineNumbers=false showCopyAction=true}
+    "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-s3/739fafa0-e4d7-11ed-a000-12d9009553ff"
+    :::
+    1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
+    ::::
```
Comment (cjhar) [inline content/intermediate/operations/drift-detection/index.md:243]:
```suggestion
1. Once the stack is created, select the `drift-detection-challenge` stack and choose **Resources**. Take a note of the **Physical ID** for `Instance1`, for example `i-1234567890abcdef0`.
    ::::
```

**Review by cjhar — COMMENTED** [review]
Thanks Shubham, I've added some suggestions for things to fix. We've also standardised the stack name and "show copy button" for stdout.
You will need to fix the stack names in the **Clean Up** section too!

**Review by cjhar — CHANGES_REQUESTED** [review]
Thanks, still two places where showCopyAction=true needs update on lines 170 and 239
```
:::code{language=shell showLineNumbers=false showCopyAction=false}
```

**Review by cjhar — DISMISSED** [review]

File: content/intermediate/operations/drift-detection/index.md [hunk 24]
```diff
@@ -224,32 +289,28 @@ echo "IyEvdXNyL2Jpbi9lbnYgYmFzaAplY2hvIEhlbGxvIFdvcmxkCg==" | base64 -d
...
-3. Once the stack update is complete, edit the template file again to remove the whole resource declaration (you can also choose to comment it out, using the `#` character at the start of each relevant line), and save the file.
-4. Update the stack with the updated template file. CloudFormation will remove the instance from the stack without terminating it.
-5. Edit the template file to restore the resource, and update the UserData to match the change made previously.
-
-```yaml
+1. Update the stack with the updated `drift-detection-challenge.yaml` template. This tells CloudFormation that when the resource is removed from the template, it should not delete it but just stop managing it.
+1. Once the stack update is complete, edit the template file again to remove the whole resource declaration (you can also choose to comment it out, using the `#` character at the start of each relevant line), and save the file.
+1. Update the stack with the updated template file. CloudFormation will remove the instance from the stack without terminating it.
```
Comment (rezabekf) [inline content/intermediate/operations/drift-detection/index.md:311]: Missing Cloud 9 instructions for updating the stack

File: content/intermediate/operations/drift-detection/index.md [hunk 25]
```diff
@@ -260,32 +321,30 @@ Resources:
...
-12. Once the stack operation is complete and the resource is imported, you can run drift detection on the stack to confirm the instance is now in sync with the stack template.
-
 :::
+1. Select the stack in the CloudFormation console, then from **Stack actions** choose **Import resources in to stack**.
+1. Choose **Next**.
+1. Upload the template file.
+1. Enter the physical ID of the instance and choose **Next**.
+1. In **Specify stack options**, choose **Next**.
+1. Choose **Import resources**.
+1. Once the stack operation is complete and the resource is imported, you can run drift detection on the stack to confirm the instance is now in sync with the stack template.
```
Comment (rezabekf) [inline content/intermediate/operations/drift-detection/index.md:331]: Missing Cloud 9 instructions for updating the stack

**Review by rezabekf — CHANGES_REQUESTED** [review]
The challenge section is missing Cloud 9 instructions, please update. Thanks.

**Review by cjhar — COMMENTED** [review]
Thanks Avinash, only a couple of small issues

**Review by cjhar — COMMENTED** [review]
Thanks LGTM

**Review by rezabekf — APPROVED** [review]
Tough one...great work!

**Pull request merged**